arabic-writing-practice-app/handwriting_test.py

```
@@ -0,0 +1,144 @@
+            else:
+                st.error(f"Unsupported image format with shape {image.shape}")
+                return ""
+        else:
+            st.error(f"Unsupported image type: {type(image)}")
+            return ""
+
+        # Convert to grayscale if not already
+        if len(img.shape) == 3:
+            gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        else:
+            gray = img
```

Ruff (0.8.2)

Use ternary operator `gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY) if len(img.shape) == 3 else img` instead of `if`-`else`-block

Replace `if`-`else`-block with `gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY) if len(img.shape) == 3 else img`

(SIM108)